lib/temporal/errors.rb

```diff
@@ -19,6 +19,9 @@ class TimeoutError < ClientError; end
   # with the intent to propagate to a workflow
   class ActivityException < ClientError; end
 
+  # Represents cancellation of a non-started activity
```

Comment:
Since cancels can also happen at a heartbeat, should we unify this case with that case?  It's not ideal that people would have to deal with both depending on the timing of when the cancellation comes in.
If that's not possible, should name it as `ActivityCanceledBeforeStart`?

Reply:
There's nothing preventing other to re-use the `Temporal::ActivityCancelled` error for heartbeat cancellations. It's just that a heartbeat cancellation is no different from any other failure raised by an activity. I think a more interesting question here is — should we provide a `#cancel!` API for the activity context that would raise that error?

Reply:
Oh, I forgot that in this SDK, heartbeat cancel doesn't throw, it returns a boolean.
In our wrapper SDK (and in the java SDK), it will throw.  We think this is the best way to make sure developers don't drop cancellations.